doc/devguide/codebase/code-style.rst

```diff
@@ -6,12 +6,165 @@ Suricata uses a fairly strict coding style. This document describes it.
+following command will format only the code changed in the most recent commit:
+
+.. code-block:: bash
+
+    $ git clang-format HEAD~1 # or HEAD^
+
+Note that this modifies the files, but doesn’t commit them – you’ll likely want to run
+
+.. code-block:: bash
+
+    $ git commit --amend -a
+
+in order to update the last commit with all pending changes.
+
+Use make targets
+""""""""""""""""
+Make targets are provided for easier usage, especially if you have multiple
+commits to format.
+
+Check if you branch changes' formatting is correct with:
+
+.. code-block:: bash
+
+    $ make check-style-branch
+
+Format changes in git staging and unstaged changes, i.e. code you are currently
+editing.
+
+.. code-block:: bash
+
+    $ make style
+    # Or with git clang-format
+    $ git clang-format --force
+
+Format every commits in your branch off master and rewrite history using the
+existing commit metadata. Tip: Create a new version of your branch first and run
+this off the new version to have the existing branch around "just in case".
+
+.. code-block:: bash
+
+    $ make style-rewrite-branch
+
+Format all changes in your branch. Allows you to add the formatting as an
+additional commit "at the end". Prefer to use make style-rewrite-branch instead.
+
+.. code-block:: bash
+
+    $ make style-branch
+
+Formatting a whole file
+"""""""""""""""""""""""
+
+.. Argh, github does not support admonitions such as .. note::
+
++--------------------------------------------------------------------+
+| **Note**                                                           |
+|                                                                    |
+| Do not reformat whole files by default, i.e. do not use            |
+| clang-format proper in general.                                    |
++--------------------------------------------------------------------+
+
+If you were ever to do so, formatting changes of existing code with clang-format
+shall be a different commit and must not be mixed with actual code changes.
+
+.. code-block:: bash
+
+    $ clang-format -i {file}
+
+Disabling clang-format
+**********************
+
+There might be times, where the clang-format's formatting might not please.
+This might mostly happen with macros, arrays (single or multi-dimensional ones),
+struct initialization, or where one manually formatted code.
+
+You can always disable clang-format.
+
+.. code-block:: c
+
+    /* clang-format off */
+    #define APP_LAYER_INCOMPLETE(c, n) (AppLayerResult){1, (c), (n)}
+    /* clang-format on */
+
+Installing clang-format and git-clang-format
+********************************************
+clang-format 9 is required.
+
+TODO - REMOVEME: We could possibly provide an "install-clang-format" script if we wanted to.
+It's essentially a decision of maintaining the install script vs support questions about
+"hey I have ubuntu xyz, what do I need to do?".
+
+On ubuntu 18.04:
+
+- It is sufficient to only install clang-format, e.g.
```

Comment:
these steps do not seem necessary. Ubuntu offers clang-9 and clang-10 through their own repos

Reply:
Will be fixed in v5